src/pages/LoginPage/LoginPage.tsx

```diff
@@ -0,0 +1,306 @@
+    if (alt === 'Google' && img.src !== GoogleIcon) {
+      img.src = GoogleIcon ?? '';
+    } else if (alt === 'GitHub' && img.src !== GithubIcon) {
+      img.src = GithubIcon ?? '';
+    } else if (alt === 'Facebook' && img.src !== FacebookIcon) {
+      img.src = FacebookIcon ?? '';
+    }
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    const newErrors = { emailOrUsername: '', password: '' };
+
+    if (!emailOrUsername) {
+      newErrors.emailOrUsername = 'Email or Username is required.';
+    } else if (emailOrUsername.includes('@') && !EMAIL_REGEX.test(emailOrUsername)) {
+      newErrors.emailOrUsername = 'Please enter a valid email address.';
+    }
+
+    if (!password) {
+      newErrors.password = 'Password is required.';
+    } else if (password.length < PASSWORD_MIN_LENGTH) {
+      newErrors.password = `Password must be at least ${PASSWORD_MIN_LENGTH} characters long.`;
+    }
+
+    if (newErrors.emailOrUsername || newErrors.password) {
+      setErrors(newErrors);
+      return;
+    }
+
+    setErrors(newErrors);
+    onSubmit({ emailOrUsername, password });
+  };
+
+  const isDisabled = disabled || loading || !!errors.emailOrUsername || !!errors.password;
+
+  return (
+    <Box sx={getPageContainerStyles()}>
+      <Box component="form" onSubmit={handleSubmit} sx={getFormStyles(theme)}>
+        <Box sx={getLogoStyles(theme)}>
+          <Logo orientation="horizontal" width="100%" height="100%" alt="Logo" />
+        </Box>
+
+        <FormInput
+          label="Mail Id"
+          value={emailOrUsername}
+          onChange={(e) => {
+            setEmailOrUsername(e.target.value);
+            setErrors({ ...errors, emailOrUsername: '' });
+          }}
+          placeholder="Mail Id"
+          errorMessage={errors.emailOrUsername}
+          sx={getInputStyles(GEOMETRY_PX.EMAIL_USERNAME_INPUT_TOP, theme)}
+        />
+        {errors.emailOrUsername && (
+          <Typography
+            variant="caption"
+            sx={getErrorTextStyles(GEOMETRY_PX.EMAIL_USERNAME_ERROR_TOP, theme)}
+          >
+            {errors.emailOrUsername}
+          </Typography>
+        )}
+
+        <FormInput
+          label="Password"
+          type="password"
+          value={password}
+          onChange={(e) => {
+            setPassword(e.target.value);
+            setErrors({ ...errors, password: '' });
+          }}
+          placeholder="Password"
+          errorMessage={errors.password}
+          sx={getInputStyles(GEOMETRY_PX.PASSWORD_INPUT_TOP, theme)}
+        />
+        {errors.password && (
+          <Typography
+            variant="caption"
+            sx={getErrorTextStyles(GEOMETRY_PX.PASSWORD_ERROR_TOP, theme)}
+          >
+            {errors.password}
+          </Typography>
+        )}
+
+        <PrimaryButton
+          sx={{
+            ...getButtonStyles(theme),
+            // REFACTOR: Usar tokens del tema en lugar de colores harcodeados
```

Comment:
Inline sx overrides use an '!important' string and reference theme.palette.grey[900], which is not defined in the login theme (login-theme defines grey[500] only). Using '!important' and undefined tokens is fragile and violates the rule to use theme tokens instead of hardcoded colors.

**Sugerencia:**
```typescript
Remove '!important' and use semantic theme tokens that exist. Example: backgroundColor: theme.palette.primary.main (or theme.palette.common.black if valid) and &:hover: { backgroundColor: theme.palette.primary.dark }. Also ensure the token exists in login-theme. Avoid '!important'.
```